### PR TITLE
 Add `contentType` in form-param

### PR DESCRIPTION
**Pull request by codenirvana**

### CODE REVIEW

File: lib/collection/form-param.js [hunk 1]
```diff
@@ -23,6 +23,7 @@ _.inherit((
         this.value = _.get(options, 'value') || '';
         this.type = _.get(options, 'type');
         this.src = _.get(options, 'src');
+        this.contentType = _.get(options, 'contentType');
```
Comment (kunagpal) [inline lib/collection/form-param.js]: While this is fine for now, can you add a todo before the first `_.get` call to get rid of all these `_.get` calls?

Comment (kunagpal) [reply lib/collection/form-param.js]: Also, shouldn't we validate/sanitize the `contentType` value first?

File: test/unit/form-param.test.js [hunk 2]
```diff
@@ -22,4 +22,27 @@ describe('FormParam', function () {
...
+                value: '',
+                src: 'fileSrc',
+                contentType: 'text/csv',
+                type: 'file'
+            });
+        });
+
+        it('should default to undefined if contentType is not provided', function () {
+            var fp = new FormParam({ key: 'foo', value: 'bar' });
+            expect(fp.toJSON()).to.eql({ key: 'foo', value: 'bar' });
```
Comment (kunagpal) [inline test/unit/form-param.test.js]: The assertion should explicitly check for `contentType` being `undefined`.

File: test/unit/form-param.test.js [hunk 3]
```diff
@@ -22,4 +22,27 @@ describe('FormParam', function () {
...
+
+    describe('contentType', function () {
+        it('should parse contentType properly', function () {
+            var fp = new FormParam({
+                key: 'fileName',
+                src: 'fileSrc',
+                contentType: 'text/csv',
+                type: 'file'
+            });
+            expect(fp.toJSON()).to.eql({
```
Comment (kunagpal) [inline test/unit/form-param.test.js]: Newline after `var`.

File: test/unit/form-param.test.js [hunk 4]
```diff
@@ -22,4 +22,27 @@ describe('FormParam', function () {
...
+                key: 'fileName',
+                value: '',
+                src: 'fileSrc',
+                contentType: 'text/csv',
```
Comment (kunagpal) [inline test/unit/form-param.test.js]: The test should only check for the existence of `contentType` in the JSON result.

File: lib/collection/form-param.js (hunk 1 above)
Comment (codenirvana) [reply lib/collection/form-param.js]: we should accept any standard/custom `content-type`.

File: test/unit/form-param.test.js (hunk 2 above)
Comment (codenirvana) [reply test/unit/form-param.test.js]: `.eql` explicitly matches the expected object. There will be no `contentType` key.

File: test/unit/form-param.test.js (hunk 3 above)
Comment (codenirvana) [reply test/unit/form-param.test.js]: following the convention used for rest of the tests.

**Pull request merged**